src/www/app/app.ts

```diff
@@ -25,32 +25,7 @@ import {OutlineServer, OutlineServerRepository} from './outline_server';
 import {Settings, SettingsKey} from './settings';
 import {Updater} from './updater';
 import {UrlInterceptor} from './url_interceptor';
-
-// If s is a URL whose fragment contains a Shadowsocks URL then return that Shadowsocks URL,
-// otherwise return s.
-export function unwrapInvite(s: string): string {
-  try {
-    const url = new URL(s);
-    if (url.hash) {
-      const decodedFragment = decodeURIComponent(url.hash);
-
-      // Search in the fragment for ss:// for two reasons:
-      //  - URL.hash includes the leading # (what).
-      //  - When a user opens invite.html#ENCODEDSSURL in their browser, the website (currently)
-      //    redirects to invite.html#/en/invite/ENCODEDSSURL. Since copying that redirected URL
-      //    seems like a reasonable thing to do, let's support those URLs too.
-      const possibleShadowsocksUrl = decodedFragment.substring(decodedFragment.indexOf('ss://'));
-
-      if (new URL(possibleShadowsocksUrl).protocol === 'ss:') {
-        return possibleShadowsocksUrl;
-      }
-    }
-  } catch (e) {
-    // Something wasn't a URL, or it couldn't be decoded - no problem, people put all kinds of
-    // unexpected things in the clipboard.
-  }
-  return s;
-}
+import {unwrapInvite} from './unwrap_invite';
```

Comment:
I had to break this out, otherwise webpack was pulling in polymer, causing all sorts of problems. If we want to snapshot components down the line we'll need to address that.

Reply:
I did not see anything referencing Polymer in this function. Why is webpack pulling polymer here?

Reply:
There are a couple ui_components being pulled in, which subsequently pulls in polymer.

Reply:
Could you clarify? I don't see any UI dependency in the code you moved.

Reply:
Oh, the problem is that the test is pulling app.ts, and we can't test app?
What's the problem with pulling Polymer? I'd rather be able to test anything in the app.ts file.

Reply:
It does stuff directly off `window` for example, which node doesn't have access to.

I can try to solve these issues, it just felt like a rabbit hole.

Reply:
We talked offline. Since this is web code, let's run the tests on a web platform. We already do that in the Manager with Karma, so we can copy: https://github.com/Jigsaw-Code/outline-server/blob/master/src/server_manager/web_app/test.action.sh.

Then we won't have to pull unrap_invite.